api/v1alpha3/doc.go

```diff
@@ -16,4 +16,6 @@ limitations under the License.
 
 // Package v1alpha3 contains the v1alpha3 API implementation.
 // +k8s:conversion-gen=sigs.k8s.io/cluster-api/api/v1beta1
+//
+// Deprecated: This package will be removed in one of the next releases.
```

Comment:
I did a brief audit of upstream k/k and couldn't find a single instance of this "deprecated package" pattern. Are we sure it's necessary?

Reply:
I think they are also not deprecating API types of older apiVersions before they drop them, right?

I could just observe that flagging the package as deprecated lead to staticcheck findings.
So I just thought it's helpful to our consumers that they get a hint that they should move away from those packages.


I think technically we have to neither deprecate the API packages nor the API types. It was just meant to help consumers overall to notice the upcomding API removal (and I wanted to avoid spending a few hours on flagging every single type in those packages as deprecated)

Reply:
is there any way of using conversion webhooks to print a warning message when the old types are used?

Reply:
You mean in the CAPI controller logs?

This could lead to a lot of spam as conversion webhooks are really called a lot

Reply:
yeah just thinking out loud if there's a more effective way to "warn" users